CabalHelper/Compile.hs

```diff
@@ -413,6 +413,9 @@ errorInstallCabal cabalVer _distdir = panic $ printf "\
  where
    sver = showVersion cabalVer
 
+helperPrepared :: Version -> IO Bool
+helperPrepared compCabalVersion = isJust <$> cachedExe compCabalVersion
```

Comment:
Maybe refactor `compileHelper` to return something like `IO (FilePath, IO (Maybe ExitCode))` where `snd` actually performs the compilation and the outer action just figures out the path to the resulting exe. Though you might not even need the outer ?`IO`.

The `MaybeT` stuff is going to need some rearanging for this I think, I'm happy to help brainstorm how to do this nicely. I don't want the code to become significatly more convoluted because of this change.